src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -591,33 +590,40 @@ private static IntermediateNode BlankToEmptyString(IntermediateNode arg)
 
                 // We are not using coalesce, because coalesce doesn't considers empty string as blank.
                 return new UnaryOpNode(convertedIRContext, UnaryOpKind.BlankToEmptyString, arg);
-            }
-
-            /// <summary>
-            /// This is not a generic arg concatenate function, but a special case for the Concatenate function,
-            /// used for Binary Concatenate operator.
-            /// </summary>
-            private static IntermediateNode ConcatenateArgs(IntermediateNode arg1, IntermediateNode arg2, IRContext irContext)
+            }
+
+#pragma warning disable CS1570 // XML comment has badly formed XML
+            /// <summary>
+            /// Some chain of binary operator (&&, &, ||) can be translated into a single call.
+            /// This unwraps the args from a child call to the parent call.
+            /// </summary>
+            /// <param name="func"></param>
+            /// <param name="arg1"></param>
+            /// <param name="arg2"></param>
+            /// <param name="irContext"></param>
+            /// <returns></returns>
+            private static IntermediateNode UnpackArgs(TexlFunction func, IntermediateNode arg1, IntermediateNode arg2, IRContext irContext)
+#pragma warning restore CS1570 // XML comment has badly formed XML
             {
-                var concatenateArgs = new List<IntermediateNode>();
+                var rangeArgs = new List<IntermediateNode>();
                 foreach (var arg in new[] { arg1, arg2 })
                 {
-                    // if arg is call node to Concatenate unpack it, and pass it as arg to outer Concatenate
-                    if (arg is CallNode maybeConcatenate && maybeConcatenate.Function is ConcatenateFunction concatenateFunction)
+                    // If arg is a call node to the same function, unpack the args.
+                    if (arg is CallNode callNode && callNode.Function.Name == func.Name)
```

Comment:
>callNode.Function.Name == func.Name [](http://example.com/codeflow?start=52&length=35)

Two functions can have the same name if they are in different namespaces. We should do a stricter comparison in this case.

Reply:
Yes, but we only call this for And/Or/Concatenate.